### ISSUE TITLE
BUG: sum method of a NA-filled Series returns zero instead of NA

### ISSUE THREAD

**Issue by prcastro** [pandas-dev/pandas#17314]
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd
import numpy as np

x = pd.Series([np.nan, np.nan])
x.sum()  # 0
```
#### Problem description

The [documentation](https://pandas.pydata.org/pandas-docs/stable/generated/pandas.Series.sum.html) states: "skipna : boolean, default True. Exclude NA/null values. If an entire row/column is NA, the result will be NA". However, the sum returns zero.

Maybe this is the intended behavior, and the problem is with the documentation, but I'm not sure.

#### Expected Output

```python
x.sum()  # np.nan
```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.2.final.0
python-bits: 64
OS: Linux
OS-release: 3.10.0-514.26.2.el7.x86_64
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.20.3
pytest: 3.2.1
pip: 9.0.1
setuptools: 27.2.0
Cython: 0.26
numpy: 1.12.1
scipy: 0.19.1
xarray: None
IPython: 6.1.0
sphinx: 1.6.3
patsy: 0.4.1
dateutil: 2.6.1
pytz: 2017.2
blosc: None
bottleneck: 1.2.1
tables: 3.4.2
numexpr: 2.6.2
feather: None
matplotlib: 2.0.2
openpyxl: 2.4.8
xlrd: 1.0.0
xlwt: 1.2.0
xlsxwriter: 0.9.8
lxml: 3.8.0
bs4: 4.6.0
html5lib: 0.9999999
sqlalchemy: 1.1.13
pymysql: None
psycopg2: None
jinja2: 2.9.6
s3fs: None
pandas_gbq: None
pandas_datareader: None

</details>

**Comment by gfyoung** [comment]
Yikes!  Something indeed doesn't match here.  IMO, `NA` is semantically different from `0`, so I would go with returning `NA` to match the docs.  PR to patch is welcome!

**Comment by TomAugspurger** [comment]
See https://github.com/pandas-dev/pandas/issues/9422 for a long discussion.

This is slated to be fixed for 0.21.
